hydromt_wflow/wflow.py

```diff
@@ -1806,11 +1806,21 @@ def setup_gauges(
                 code = self.crs.to_epsg()
             else:
                 code = self.crs
-            kwargs.update(crs=code)
+            # Add crs to metadata for DataSource construction
+            if "metadata" in kwargs:
+                kwargs["metadata"].update(crs=code)
+            else:
+                kwargs.update(metadata={"crs": code})
+
+            # Add driver argument when file is a table
+            ext = Path(gauges_fn).suffix
+            if ext in [".csv", ".parquet", ".xls", ".xlsx"]:
+                driver_dict = {"driver": {"name": "geodataframe_table"}}
+                kwargs.update(driver_dict)
```

Comment:
Nice! I feel like actually this should be merged with the crs. Ie we only want to modify the CRS for if the file extension is csv etc. and not geojson, shp. And then you can remove the comment # hydromt#1243 above :)